diff --git a/server/etcdserver/api/v3rpc/watch.go b/server/etcdserver/api/v3rpc/watch.go
index cd834aa..e6aaf2b 100644
--- a/server/etcdserver/api/v3rpc/watch.go
+++ b/server/etcdserver/api/v3rpc/watch.go
@@ -460,6 +460,7 @@ func (sws *serverWatchStream) sendLoop() {
                        sws.mu.RUnlock()

                        var serr error
+                       // gofail: var beforeSendWatchResponse struct{}
                        if !fragmented && !ok {
                                serr = sws.gRPCStream.Send(wr)
                        } else {
